src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -55,10 +68,12 @@ public Task updateTask(Task task, Long taskId) {
         Task checkedTask = taskRepository.getOne(taskId);
         if (!checkedTask.getUserId().equals(user.getId())) {
             throw new AccessDeniedException("User with login " + user.getId() + " cannot edit task with id " + taskId);
+        } else if (checkedTask.getPlannedFor().isBefore(LocalDateTime.now())) {
+            throw new TaskExpiredException("Time for finish task with id " + checkedTask.getId() + " had expired at: " + checkedTask.getPlannedFor());
```

Comment:
This doesn't add up. Planned for should be a date, not time. That's first thing, second one is that planned for is date/time when the task was planned for, not when it expires, but this exception message implies something else.

Reply:
It's logic that when planned for date is past then task is expired, how said it differently ? Btw I will change message to : "Task with id: (task.id) was planned for: (task.plannedFor).You cannnot modify this task any more."

Reply:
Field name says `plannedFor` not `expiresAt` that's a huge difference.